tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -0,0 +1,1078 @@
+                resolvedField.getName() +
+                ";").trim();
+    }
+
+    /**
+     * Returns the signature of a JavaParser callable declaration. Uses the
+     * method source code and removes method body, contained comments, the
+     * Javadoc comment, and other special characters (e.g. "\n").
+     *
+     * @param jpCallable a JavaParser callable declaration
+     * @return a string representation of the signature. A signature follows
+     * the format:
+     *     "[modifiers] [typeParameters] [type] [methodName]([parameters]) throws [exceptions]"
+     */
+    public static String getCallableSignature(
+            CallableDeclaration<?> jpCallable
+    ) {
+        // get method source as a string
+        String methodSignature = jpCallable.toString();
+        Optional<BlockStmt> methodBody = jpCallable instanceof MethodDeclaration ?
+                ((MethodDeclaration) jpCallable).getBody() :
+                Optional.ofNullable(((ConstructorDeclaration) jpCallable).getBody());
+        // remove method body
+        if (methodBody.isPresent()) {
+            methodSignature = methodSignature.replace(methodBody.get().toString(), "");
+        }
+        // remove all comments
+        for (Node comment: jpCallable.getAllContainedComments()) {
+            methodSignature = methodSignature.replace(comment.toString(), "");
+        }
+        // remove Javadoc comment
+        if (jpCallable.getComment().isPresent()) {
+            methodSignature = methodSignature.replaceAll("[\\s\\S]*\n", "");
+        }
+        // remove special characters
+        methodSignature = methodSignature.replaceAll("/\\*\\*([\\s\\S]*?)\\*/(\\n|\\r|\\t)*", "");
+        return methodSignature.trim().replaceAll(";$", "");
+    }
+
+    /**
+     * Returns the signature of a JavaParser method declaration.
+     *
+     * @param methodDeclaration a JavaParser method declaration
+     * @return a string representation of the signature. Signature follows the
+     * format:
+     *  "[modifiers] [type] [methodName]([parameters]) throws [exceptions]"
+     */
+    public static String getMethodSignature(MethodDeclaration methodDeclaration) {
+        String method = methodDeclaration.toString();
+        if (methodDeclaration.getBody().isPresent()) {
+            // Remove body
+            method = method.replace(methodDeclaration.getBody().get().toString(), "");
+        }
+        for (Node comment: methodDeclaration.getAllContainedComments()) {
+            // Remove comments within method signature
+            method = method.replace(comment.toString(), "");
+        }
+        if (methodDeclaration.getComment().isPresent()) {
+            // At this point, last line is method signature. Remove everything before that
+            method = method.replaceAll("[\\s\\S]*\n", "");
+        }
+        return method.trim().replaceAll(";$", "");
+    }
+
+    /**
+     * Gets all type parameters of a given method.
```

Comment:
In what format?  As simple names?  As declared in the source code, which would include bounds?